src/wallet/asyncrpcoperation_shieldcoinbase.cpp

```diff
@@ -93,8 +93,8 @@ AsyncRPCOperation_shieldcoinbase::AsyncRPCOperation_shieldcoinbase(
         [&](libzcash::SproutPaymentAddress addr) {
             tozaddr_ = addr;
         },
-        [&](libzcash::UnifiedAddress) {
-            throw JSONRPCError(RPC_VERIFY_REJECTED, "Cannot shield coinbase output to a unified address.");
+        [&](libzcash::UnifiedAddress addr) {
+            tozaddr_ = addr;
```

Comment:
This is insufficient to add Unified Address support; we also need to implement the necessary logic in `ShieldToAddress::operator()(const libzcash::UnifiedAddress &uaddr)`. What that should do is look through the UA's recipients in preference order (which currently is what `for (const auto& receiver : uaddr)` will produce), and then send to the first one we are able to. Currently this means skipping an Orchard receiver if present, sending to Sapling if present, and then otherwise failing (since no Sapling address would mean the UA only contains Orchard and transparent, which we can't currently handle). The "send to Sapling" part can be easily handled by calling `ShieldToAddress(m_op, sendAmount)(saplingAddr)` from within the UA handler.